cms/datasets/views.py

```diff
@@ -104,10 +177,13 @@ def get_objects(self, pks: list[Any]) -> QuerySet[Dataset]:
 
         # TODO: update when we can fetch items in bulk from the dataset API or use the cached listing view?
         for pk in pks:
-            item_from_api = ONSDataset.objects.get(pk=pk)  # pylint: disable=no-member
+            # The provided PK is actually a combination of dataset_id/edition/version
+            dataset_id, edition, version = deconstruct_dataset_compound_id(str(pk))
+
+            item_from_api = ONSDataset.objects.get(pk=dataset_id)  # pylint: disable=no-member
 
             api_data_for_datasets.append(item_from_api)
```

Comment:
Adding the item from the api as is, can make this fail as we are querying by just the dataset id, we need to do the same as the other single chooser, only use the item from api for title/description.

This line would fail because `data.version` and the existing version may not match, as `data.version` returns the latest version.
```
for data in api_data_for_datasets:
	data_version = int(data.version)
	if (data.id, data.edition, data_version) not in existing_datasets_map: